src/game_of_life.test.js

```diff
@@ -0,0 +1,156 @@
+import next, { countLiveNeighbours } from './game_of_life';
+
+describe('Game of life', () => {
+  test('Function next exists', () => {
+    expect(typeof next).toBe('function');
+  });
+
+  test('next returns array of array', () => {
+    const input = [[]];
+    const output = next(input);
+    expect(Array.isArray(output)).toBe(true);
+    expect(Array.isArray(output[0])).toBe(true);
+  });
+
+  test('one live cell will die', () => {
+    const input = [[1]];
+    const expectedOutput = [[0]];
+    const output = next(input);
+    expect(output).toEqual(expectedOutput);
+  });
+
+  test('underpopulation', () => {
+    const input = [
+      [0, 1, 1],
+      [0, 0, 0],
+      [1, 1, 0],
+    ];
+
+    const output = next(input);
+    expect(output[0][1]).toEqual(0);
+    expect(output[0][2]).toEqual(0);
+    expect(output[2][0]).toEqual(0);
+    expect(output[2][1]).toEqual(0);
+  });
+
+  test('overcrowding', () => {
+    const input = [
+      [0, 1, 1],
+      [1, 1, 1],
+      [0, 0, 0],
+    ];
+    const output = next(input);
+    expect(output[1][1]).toEqual(0);
+  });
+
+  test('live cell survives with 2 or 3 neighbours', () => {
+    const input = [
+      [0, 1, 0],
+      [0, 1, 0],
+      [1, 1, 0],
+    ];
+    const output = next(input);
+    expect(output[0][2]).toEqual(0);
+    expect(output[2][0]).toEqual(1);
+    expect(output[1][1]).toEqual(1);
+  });
+
+  test('cell is born with exactly three neighbours', () => {
+    const input = [
+      [0, 1, 0],
+      [0, 1, 1],
+      [0, 0, 0],
+    ];
+    const output = next(input);
+    expect(output[0][2]).toEqual(1);
+  });
+
+  test('given example from the webpage', () => {
+    const input = [
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 0, 1, 0, 0, 0],
+      [0, 0, 0, 1, 1, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+    ];
+    const expectedOutput = [
+      [0, 0, 0, 0, 0, 0, 0, 0],
+      [0, 0, 0, 1, 1, 0, 0, 0],
+      [0, 0, 0, 1, 1, 0, 0, 0],
+      [0, 0, 0, 0, 0, 0, 0, 0],
+    ];
+    const output = next(input);
+    expect(output).toEqual(expectedOutput);
+  });
+
+  test('still alive', () => {
+    const input = [
+      [0, 0, 1, 1, 0, 0],
+      [0, 1, 0, 0, 1, 0],
+      [0, 1, 0, 0, 1, 0],
+      [0, 0, 1, 1, 0, 0],
+    ];
+
+    const output = next(input);
+    expect(output).toEqual(input);
+  });
+});
+
+describe('countLiveNeighbours', () => {
+  test('all cells are alive', () => {
+    const input = [
+      [1, 1, 1],
+      [1, 1, 1],
+      [1, 1, 1],
+    ];
+
+    expect(countLiveNeighbours(input, 0, 0)).toBe(3);
```

Comment:
Is this really a good practice to use multiple expect in one test? I'm always feeling like if this fail you will not know what exactly is wrong and you need to go and investigate deeper.

Reply:
Yes, it's not ideal. But I really wanted to test all position.